### PR TITLE
HDFS wire encryption

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes #8604

File: presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHiveMetastoreAuthentication.java [hunk 1]
```diff
@@ -60,7 +66,7 @@ public TTransport authenticate(TTransport rawTransport, String hiveMetastoreHost
                     "Kerberos principal name does NOT have the expected hostname part: %s", serverPrincipal);
 
             Map<String, String> saslProps = ImmutableMap.of(
-                    Sasl.QOP, "auth",
+                    Sasl.QOP, hdfsWireEncryptionEnabled ? "auth-conf" : "auth",
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHiveMetastoreAuthentication.java]: Do you think we should have a separate property for metastore traffic encryption? e.g.: `hive.metastore.wire-encryption.enabled`

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java [hunk 2]
```diff
@@ -954,6 +955,19 @@ public HiveClientConfig setHdfsImpersonationEnabled(boolean hdfsImpersonationEna
         return this;
     }
 
+    public boolean isHdfsWireEncryptionEnabled()
+    {
+        return hdfsWireEncryptionEnabled;
+    }
+
+    @Config("hive.hdfs-wire-encryption.enabled")
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java]: How about `hive.hdfs.wire-encryption.enabled`?

File: .travis.yml [hunk 3]
```diff
@@ -96,6 +96,11 @@ script:
       presto-product-tests/bin/run_on_docker.sh \
         singlenode-kerberos-hdfs-no-impersonation -g hdfs_no_impersonation
     fi
+  - |
+    if [[ -v PRODUCT_TESTS_SPECIFIC_ENVIRONMENT ]]; then
```
Comment (arhimondr) [inline .travis.yml]: Consider `PRODUCT_TESTS_SPECIFIC_ENVIRONMENT_2`, `PRODUCT_TESTS_SPECIFIC_ENVIRONMENT` already takes more time.

File: presto-docs/src/main/sphinx/connector/hive-security.rst [hunk 4]
```diff
@@ -499,3 +499,20 @@ See below for an example.
       ]
     }
 
+HDFS wire encryption
+--------------------
+
+In a Kerberized Hadoop cluster with enabled HDFS wire encryption you can enable
```
Comment (arhimondr) [inline presto-docs/src/main/sphinx/connector/hive-security.rst]: Please ask someone else to review this commit. @mattsfuller ?

File: presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+-->
+<configuration>
+
+    <property>
+        <name>fs.defaultFS</name>
+        <value>hdfs://hadoop-master:8020</value>
+    </property>
+
+    <property>
+        <name>fs.default.name</name>
```
Comment (findepi) [inline presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml]: `fs.default.name` or `fs.defaultFS` is obsolete (don't remember)

File: presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        <value>hdfs://hadoop-master:9000</value>
+    </property>
+
+    <!-- OOZIE proxy user setting -->
+    <property>
+        <name>hadoop.proxyuser.oozie.hosts</name>
+        <value>*</value>
+    </property>
+    <property>
+        <name>hadoop.proxyuser.oozie.groups</name>
```
Comment (findepi) [inline presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml]: needed?

File: presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        <value>*</value>
+    </property>
+
+    <!-- Llama proxy user setting -->
+    <property>
+        <name>hadoop.proxyuser.llama.hosts</name>
+        <value>*</value>
+    </property>
+    <property>
+        <name>hadoop.proxyuser.llama.groups</name>
```
Comment (findepi) [inline presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml]: needed?

File: presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        <value>*</value>
+    </property>
+
+    <!-- Hue proxy user setting -->
+    <property>
+        <name>hadoop.proxyuser.hue.hosts</name>
+        <value>*</value>
+    </property>
+    <property>
+        <name>hadoop.proxyuser.hue.groups</name>
```
Comment (findepi) [inline presto-product-tests/conf/hadoop/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/core-site.xml]: not needed, right?

File: presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/hive.properties [hunk 9]
```diff
@@ -0,0 +1,27 @@
+#
+# WARNING
+# ^^^^^^^
+# This configuration file is for development only and should NOT be used be
+# used in production. For example configuration, see the Presto documentation.
```
Comment (findepi) [inline presto-product-tests/conf/presto/etc/environment-specific-catalogs/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/hive.properties]: We don't need this WARNING comment here.

File: presto-docs/src/main/sphinx/connector/hive-security.rst (hunk 4 above)
Comment (kokosing) [reply presto-docs/src/main/sphinx/connector/hive-security.rst]: I think @mattsfuller already reviewed this internally.

**Pull request merged**